addons/pos_self_order/static/src/ProductList/ProductList.js

```diff
@@ -0,0 +1,217 @@
+    };
+
+    scrollToTop = () => {
+        this.productListDiv?.el?.scroll({
+            top: 0,
+            left: 0,
+            behavior: "smooth",
+        });
+    };
+    /**
+     * @param {Object} element - the element to scroll to (should be a ref)
+     * @description this function scrolls the productListDiv to the element passed as argument,
+     *            it takes into account the height of the header
+     */
+    scrollToElement = (element) => {
+        const y = element?.el.offsetTop;
+        // we are working with floats, so by the time we scroll to the element, we might be a bit off
+        // so we add a couple of px to make sure that we are on the correct side of the error
+        const SCROLL_IMPRECISSION_CORRECTION = 4;
+        const scrollOffset = this.headerHeight - SCROLL_IMPRECISSION_CORRECTION;
+        this.productListDiv?.el?.scroll({
+            top: y - scrollOffset,
+            behavior: "smooth",
+        });
+    };
+    /**
+     * @returns {Object} the list of products that should be displayed
+     * @description this function returns the list of products that should be displayed;
+     *             it filters the products based on the search input
+     */
+    filteredProducts = () => {
+        if (!this.privateState.searchInput) {
+            return this.props.productList;
+        }
+        return fuzzyLookup(
+            this.privateState.searchInput,
+            this.props.productList,
+            (product) => product.name + product.description_sale
+        );
+    };
+    /**
+     * @param {string} tag_name
+     * @description this function is called when a tag is clicked; it selects the chosen tag and deselects all the other tags
+     */
+    selectTag = (tag_name) => {
+        if (this.privateState.selectedTag === tag_name) {
+            this.privateState.selectedTag = "";
+            this.scrollToTop();
+            return;
+        }
+        // When the user clicks on a tag, we scroll to the part of the page
+        // where the products with that tag are displayed.
+        // after the scrolling is done, the intersection observer will
+        // automatically set the privateState.selectedTag to the tag_name
+        this.scrollToElement(this.productGroupDiv[tag_name]);
+    };
+    /**
+     * @description This function is called when the search button is clicked.
+     * It sets the state so the search input is focused.
+     * It also deselects all the selected tags
+     */
+    focusSearch = () => {
+        this.privateState.searchIsFocused = true;
+        if (this.privateState.navbarIsShown) {
+            this.hideNavBar(true);
+        }
+        this.scrollToTop();
+    };
+    /**
+     * @description this function is called when the search input 'x' button is clicked
+     */
+    closeSearch = () => {
+        this.privateState.searchIsFocused = false;
+        this.privateState.searchInput = "";
+    };
+}
+export default { ProductList };
```

Comment:
ProductList above is already being exported. No need for this line. In general, do not use `export default`, just directly export that class/function inline.